atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/TopListFilterTest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.Duration;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.Test;
+
+public final class TopListFilterTest {
```

Comment:
Some of this relates to the `CoalescingSupplier` behaviour. I think as the tests get more complex (which they will likely become if we implement some of the other behaviour changes suggested above) we might want to split this out so that there are a few tests checking that the waiting overall makes sense, and for the rest we pass in a `CachingSupplier` or something that can be manually explicitly invalidated in the tests.